### PR TITLE
SRC-6063 Wait for trajectory topic instead of sleeping

### PR DESCRIPTION
**Pull request by mykolasjuraitis**
## Proposed changes

Sleeping for 30 seconds doesn't always seem to be enough for controllers to start. In order to not block startup when controllers do start early now scripts would wait for appearance of trajectory topics. Waiting timeout was increased to 1 minute.

## Checklist

Before posting a PR ensure that from each of the below categories **AT LEAST ONE BOX HAS BEEN CHECKED**. If more than one category is applicable then more can be checked. Also ensure that the proposed changes have been filled out with relevant information for reviewers.

## Tests

- [x] No tests required to be added. (For small changes that will be tested by CI/CD infrastructure).
- [ ] Added automated tests (if a new class is added (Python or C++), the interface of that class must be unit tested).
- [ ] Manually tested in simulation (if simulation specific or no hardware required to test the functionality). 
- [ ] Manually tested on hardware (if hardware specific or related).

## Documentation

- [x] No documentation required to be added.
- [ ] Added documentation (For any new feature, explain what it does and how to use it. Write the documentation in a relevant space, e.g. Github, Confluence, etc).
- [ ] Updated documentation (For changes to pre-existing features mentioned in the documentation).

### CODE REVIEW

File: ansible/roles/products/teleop/server/desktop-icons/templates/scripts/wait_for_topic.j2 [hunk 1]
```diff
@@ -0,0 +1,17 @@
+#jinja2: trim_blocks:False
+#!/bin/bash
+
+# Arg1: Topic name to wait for
+# Arg2: Error message to display if topic doesn't appear until timeout
+elapsed_time=0
+start_time=$SECONDS
+while [ $elapsed_time -lt 60 ]; do
```
Comment (mikramarc) [inline ansible/roles/products/teleop/server/desktop-icons/templates/scripts/wait_for_topic.j2:8]: could we make the "60" a param just for flexibility sake? Or at least move it to a "const" variable?

File: ansible/roles/products/teleop/server/desktop-icons/templates/scripts/wait_for_topic.j2 [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+  elapsed_time=$(( SECONDS - start_time ))
+  if [[ "$(docker exec -ti {{ container_name }} bash -c "source /home/user/projects/shadow_robot/base/devel/setup.bash;rostopic list")" = *"${1}"* ]]; then
+    # Success
+    exit 0
+  fi
+done
+zenity --error --text="${2}"
+docker stop {{ container_name }}
```
Comment (mikramarc) [inline ansible/roles/products/teleop/server/desktop-icons/templates/scripts/wait_for_topic.j2:16]: this kind of break the single responsibility principle, could we please move the docker stop step outside the script into the scripts that call this one? (if exit code 1 -> docker stop)

**Review by mikramarc — CHANGES_REQUESTED** [review]

File: ansible/roles/products/teleop/server/desktop-icons/templates/scripts/wait_for_topic.j2 (hunk 1 above)
Comment (mykolasjuraitis) [reply ansible/roles/products/teleop/server/desktop-icons/templates/scripts/wait_for_topic.j2]: Extracted constant in e7efe2ff5007e7e4596c4c507386c023cd712690

File: ansible/roles/products/teleop/server/desktop-icons/templates/scripts/wait_for_topic.j2 (hunk 2 above)
Comment (mykolasjuraitis) [reply ansible/roles/products/teleop/server/desktop-icons/templates/scripts/wait_for_topic.j2]: Moved in 3f7b0aafedb84a371589d2328d94e32c0e15b0dc.

**Review by mikramarc — APPROVED** [review]

**Pull request closed without merging**